compiler/optimizer/OMRCFGSimplifier.cpp

```diff
@@ -714,6 +714,47 @@ bool OMR::CFGSimplifier::simplifyNullToException(bool needToDuplicateTree)
    return true;
    }
 
+static bool checkEquivalentIndirectLoadChain(TR::Node *lhs, TR::Node *rhs)
+   {
+   if (lhs->getOpCodeValue() != rhs->getOpCodeValue())
+      return false;
+
+   if (lhs->getNumChildren() != rhs->getNumChildren())
+      return false;
+
+   if (lhs->getOpCode().hasSymbolReference()
+       && lhs->getSymbolReference()->getReferenceNumber() != rhs->getSymbolReference()->getReferenceNumber())
+      return false;
+
+   if (lhs->getOpCode().isLoadDirect())
+      {
+      return true;
+      }
+   else if (lhs->getOpCode().isLoadIndirect()
+            && lhs->getNumChildren() == 1
+            && checkEquivalentIndirectLoadChain(lhs->getFirstChild(), rhs->getFirstChild()))
+      {
+      return true;
+      }
+   else if (lhs->getOpCodeValue() == TR::aladd
+            && lhs->getSecondChild()->getOpCodeValue() == rhs->getSecondChild()->getOpCodeValue()
+            && lhs->getSecondChild()->getOpCodeValue() == TR::lconst
+            && lhs->getConstValue() == rhs->getConstValue()
+            && checkEquivalentIndirectLoadChain(lhs->getFirstChild(), rhs->getFirstChild()))
+      {
+      return true;
+      }
+   else if (lhs->getOpCodeValue() == TR::aiadd
+            && lhs->getSecondChild()->getOpCodeValue() == rhs->getSecondChild()->getOpCodeValue()
+            && lhs->getSecondChild()->getOpCodeValue() == TR::iconst
+            && lhs->getConstValue() == rhs->getConstValue()
```

Comment:
Should this be getting the constant value from the second children if lhs and rhs ?

Reply:
fixed in 27bfdfc